lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1447,6 +1470,35 @@ FIRRTLModuleLowering::lowerFormalBody(verif::FormalOp formalOp,
   return success();
 }
 
+/// Run on each `verif.simulation` to populate its body based on the original
+/// `firrtl.simulation` operation.
+LogicalResult
+FIRRTLModuleLowering::lowerSimulationBody(verif::SimulationOp simulationOp,
+                                          CircuitLoweringState &loweringState) {
+  auto builder = OpBuilder::atBlockEnd(simulationOp.getBody());
+
+  // Find the module targeted by the `firrtl.simulation` operation.
+  auto oldSimulationOp =
+      cast<SimulationOp>(loweringState.getOldModule(simulationOp));
+  auto moduleName = oldSimulationOp.getModuleNameAttr().getAttr();
+  auto oldModule = cast<FModuleLike>(
+      *loweringState.getInstanceGraph().lookup(moduleName)->getModule());
+  auto newModule =
+      dyn_cast_or_null<hw::HWModuleLike>(loweringState.getNewModule(oldModule));
+  if (!newModule)
+    return oldSimulationOp->emitOpError()
+           << "could not find module " << oldModule.getNameAttr();
```

Comment:
Can this check be removed since I _think_ it is already guaranteed by a verifier?

Reply:
That's a good point! Let me change these to casts, and make things bit less verbose.